src/Nethermind/Nethermind.Network/PeerManager.cs

```diff
@@ -380,8 +343,34 @@ private async Task RunPeerUpdateLoop()
 
                 _peerUpdateTimer.Start();
             }
+
+            taskChannel.Writer.Complete();
+            await Task.WhenAll(tasks);
         }
 
+        private bool EnsureAvailableActivePeerSlot()
+        {
+            if (AvailableActivePeersCount - _pending > 0)
+            {
+                return true;
+            }
+
+            // Once the connection was established, the active peer count will increase, but it might
+            // not pass the handshake and the status check. So we wait for a bit to see if we can get
+            // the active peer count to go down within this time window.
+            DateTimeOffset deadline = DateTimeOffset.Now + Timeouts.Handshake +
+                                      TimeSpan.FromMilliseconds(_networkConfig.ConnectTimeoutMs);
+            while (DateTimeOffset.Now < deadline && (AvailableActivePeersCount - _pending) <= 0)
```

Comment:
make it Task.Delay, instead of spinning? Use ValueTask as return value if we don't need to wait.

Reply:
Yea, but its possible that an available peer count is available before the deadline, so it need some way to exit early.  The _peerUpdateRequested is not reliable in this case.